src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.Editor.cs

```diff
@@ -293,17 +295,32 @@ private void AddFoldersToNamespaceContainers(List<string> container, IList<strin
                 // 1: folders -> if triggered from Dialog
                 // 2: containers -> if triggered not from a Dialog but from QualifiedName
                 // 3: triggering document folder structure -> if triggered not from a Dialog and a SimpleName
-                var adjustedContainer = isDialog ? folders :
-                        _state.SimpleName != _state.NameOrMemberAccessExpression ? containers.ToList() : _document.Document.Folders.ToList();
+                var adjustedContainer = isDialog 
+                    ? folders 
+                    : _state.SimpleName != _state.NameOrMemberAccessExpression
+                        ? containers.ToList()
+                        : _document.Document.Folders.ToList();
 
                 // Now, take the code that would be generated and actually create an edit that would
                 // produce a document with that code in it.
+                var newRoot = await codeGenResult.GetSyntaxRootAsync(_cancellationToken).ConfigureAwait(false);
+
+                if (newDocument.Project.Language == _document.Document.Project.Language)
+                {
+                    var syntaxFacts = _document.Document.GetLanguageService<ISyntaxFactsService>();
+                    var fileBanner = syntaxFacts.GetFileBanner(_document.Root);
+
+                    if (fileBanner.Any(syntaxFacts.IsRegularComment))
```

Comment:
Nope.  You're right.  it always has to reutrn at least one comment.  Will remove this code.